cli/cmd/cluster.go

```diff
@@ -0,0 +1,206 @@
+			"instance.yaml",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdCtx.CommandName = cmd.Name()
+			err := modules.RunCmd(&cmdCtx, cmd.CommandPath(), &modulesInfo,
+				internalClusterPublishModule, args)
+			handleCmdErr(cmd, err)
+		},
+		Args: cobra.ExactArgs(2),
+		ValidArgsFunction: func(
+			cmd *cobra.Command,
+			args []string,
+			toComplete string) ([]string, cobra.ShellCompDirective) {
+			return internal.ValidArgsFunction(
+				cliOpts, &cmdCtx, cmd, toComplete,
+				running.ExtractActiveAppNames,
+				running.ExtractActiveInstanceNames)
+		},
+	}
+	publish.Flags().BoolVar(&publishCtx.Force, "force", publishCtx.Force,
+		"force publish and skip validation")
+	clusterCmd.AddCommand(publish)
+
+	return clusterCmd
+}
+
+// internalClusterShowModule is an entrypoint for `cluster show` command.
+func internalClusterShowModule(cmdCtx *cmdcontext.CmdCtx, args []string) error {
+	if uri, err := url.Parse(args[0]); err == nil && uri.Scheme != "" {
+		return clustercmd.ShowEtcd(showCtx, uri)
+	}
+
+	// It looks like an application or an application:instance.
+	path, name, err := parseAppStr(cmdCtx, args[0])
+	if err != nil {
+		return err
+	}
+
+	return clustercmd.ShowCluster(showCtx, path, name)
+}
+
+// internalClusterPublishModule is an entrypoint for `cluster publish` command.
+func internalClusterPublishModule(cmdCtx *cmdcontext.CmdCtx, args []string) error {
+	data, config, err := readSourceFile(args[1])
+	if err != nil {
+		return err
+	}
+	publishCtx.Src = data
+	publishCtx.Config = config
+
+	if uri, err := url.Parse(args[0]); err == nil && uri.Scheme != "" {
+		return clustercmd.PublishEtcd(publishCtx, uri)
+	}
+
+	// It looks like an application or an application:instance.
+	path, name, err := parseAppStr(cmdCtx, args[0])
+	if err != nil {
+		return err
+	}
+
+	return clustercmd.PublishCluster(publishCtx, path, name)
+}
+
+// readSourceFile reads a configuration from a source file.
+func readSourceFile(path string) ([]byte, *cluster.Config, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to read path %q: %s", path, err)
+	}
+
+	config, err := cluster.NewYamlCollector(data).Collect()
+	if err != nil {
+		err = fmt.Errorf("failed to read a configuration from path %q: %s",
+			path, err)
+		return nil, nil, err
+	}
+
+	return data, config, nil
+}
+
+// parseAppStr parses a string and returns an application configuration path
+// and an application instance name or an error.
+func parseAppStr(cmdCtx *cmdcontext.CmdCtx, appStr string) (string, string, error) {
+	var (
+		runningCtx running.RunningCtx
+		name       string
+	)
+
+	if !isConfigExist(cmdCtx) {
+		return "", "", fmt.Errorf("unable to resolve the application name %q: %w",
+			appStr, errNoConfig)
+	}
+
+	err := running.FillCtx(cliOpts, cmdCtx, &runningCtx, []string{appStr})
+	if err != nil {
+		return "", "", err
+	}
+
+	path := filepath.Join(runningCtx.Instances[0].AppDir, configFileName)
```

Comment:
Let's use `ClusterConfigPath`.
`path` variable name purpose is unclear. Must be "clusterConfigPath" or something like this.

Reply:
Updated.